hudi-spark/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -297,8 +298,9 @@ object DataSourceWriteOptions {
   val DEFAULT_HIVE_ASSUME_DATE_PARTITION_OPT_VAL = "false"
   val DEFAULT_USE_PRE_APACHE_INPUT_FORMAT_OPT_VAL = "false"
   val DEFAULT_HIVE_USE_JDBC_OPT_VAL = "true"
-
   // Async Compaction - Enabled by default for MOR
   val ASYNC_COMPACT_ENABLE_KEY = "hoodie.datasource.compaction.async.enable"
   val DEFAULT_ASYNC_COMPACT_ENABLE_VAL = "true"
+  // Internal configs
```

Comment:
this is publicly visible. cannot add this here.

Reply:
cc @bvaradar @nsivabalan let us think very proactively about adding anything to public classes :) . Also I notice `INSTANT_TIME` does not follow the key naming conventions in the rest of the configs. these little things add up.

Reply:
also renaming `ASYNC_COMPACT_ENABLE_KEY` in the process